operator/v1/types_dns.go

```diff
@@ -0,0 +1,98 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DNS manages CoreDNS component to provide a name resolution service
+// in the cluster.
+type DNS struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the DNS.
+	Spec DNSSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the DNS.
+	Status DNSStatus `json:"status,omitempty"`
+}
+
+// DNSSpec is the specification of the desired behavior of the DNS.
+type DNSSpec struct {
+}
+
+// DNSStatus defines the observed status of the DNS.
+type DNSStatus struct {
+	// ClusterIP is the internal service IP reserved for cluster DNS service.
+	//
+	// Pods can use this IP to resolve domain names.
+	// This can be facilitated by one of these ways:
+	// * Enabling kubelet to inject this service IP in pod's nameserver list.
+	//   (this could be /etc/resolv.conf in some of the linux distributions)
+	// * Specify this service IP in pod's spec.dnsConfig.nameservers list.
+	// * Explicitly use this service IP as resolver.
```

Comment:
What is the reason for enumerating these ways? We always inject the service IP address into a pod's `/etc/resolv.conf` if its DNS policy is `ClusterFirst`, the default, and the API documentation for Pod covers the various options for configuring a pod's DNS.

Reply:
Only 'default' ClusterDNS (DNS now) injects service IP address in pod's resolver file as it is passed as `--cluster-ip` for kubelet. This is generic DNS api and I tried to explain how this field can be used in general.

Reply:
I see.  What do you think of the following wording?
> ClusterIP is the service IP through which this DNS is made available.
> 
> In the case of the default DNS, this will be a well known IP that is used as the default nameserver for pods that are using the default `ClusterFirst` DNS policy.
> 
> In general, this IP can be specified in a pod's `spec.dnsConfig.nameservers` list or used explicitly when performing name resolution from within the cluster.  Example: `dig foo.com @<service IP>`

Reply:
Updated, thanks